parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -500,12 +508,49 @@ loop = do
 
       DiffNamespaceI before0 after0 -> do
         let [beforep, afterp] =
-              Path.toAbsolutePath currentPath' <$> [before0, after0]
+              resolveToAbsolute <$> [before0, after0]
         before <- Branch.head <$> getAt beforep
         after <- Branch.head <$> getAt afterp
         (ppe, outputDiff) <- diffHelper before after
         respondNumbered $ ShowDiffNamespace beforep afterp ppe outputDiff
 
+      CreatePullRequestI baseRepo headRepo -> do
+        baseBranch <- loadRemoteBranch baseRepo
+        headBranch <- loadRemoteBranch headRepo
+        case (baseBranch, headBranch) of
+          (Left e1, Left e2) -> gitError e1 >> gitError e2
+          (Left e, _) -> gitError e
+          (_, Left e) -> gitError e
+          (Right baseBranch, Right headBranch) -> do
+            merged <- eval . Eval $ Branch.merge baseBranch headBranch
+            (ppe, diff) <- diffHelper (Branch.head baseBranch) (Branch.head merged)
+            respondNumbered $ ShowDiffAfterCreatePR baseRepo headRepo ppe diff
+        where
+        gitError = eval . Notify . GitError input
+
+      LoadPullRequestI baseRepo headRepo ->
+        if Branch.isEmpty0 currentBranch0 then do
+          let textToRelative = 
+                Path.Relative . Path.Path . Seq.singleton . NameSegment.NameSegment
+              base = textToRelative "base"
```

Comment:
I wonder how builtin strings like this will interact with the pluggable syntax; I guess there'd just be a partial function pulled out of the sky to parse a `Text` as a `Maybe NameSegment`, and if `"base"` is an invalid name segment... then you can't load a pull request 😅

Reply:
Hmm, I guess I'd think "base" here would be the literal underlying name, and that any pluggable syntax should have some way to represent it (e.g. syntax-specific escaping).

Like, we should probably allow a namespace called `.where`; we currently don't but it's because the ucm command parser is inappropriately (IMO) sharing the basic unison syntax parser for wordyIds.

Reply:
Ah, I see... so then there would be global rules for what a name segment _is_, i.e. maybe it's just defined as a non-empty string of unicode code points

Reply:
Yeah, I think it's that.

Reply:
In that case I think it's safe/smart to put an `IsString` instance for relative paths (which creates a singleton path of course)

Reply:
I worry about accidentally trying to create a non-singleton path from a string constant and ending up with an unexpected singleton path; WDYT?  NameSegment currently has an `IsString` instance though.  Maybe a constructor like `Relative.singleton` though?

Reply:
Yeah; `Relative.singleton` is a good idea. I do think you're right about the confusion since the codebase today does have strings like `"Int.+"` that are turned into a multi-segment name. But when the concept of pluggable syntax matures, I think it will become more be intuitive/obvious that `.` is not special in any way, so a string literal really cannot be multi-segment.

Reply:
Also I forgot that there was an `instance IsString NameSegment`.  I got rid of the call to `textToRelative` and added a call to `relativeSingleton :: NameSegment -> Relative` using the `IsString`.  Thanks for pointing it out!